src/validation_workflow/rpm/validation_rpm.py

```diff
@@ -0,0 +1,76 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import time
+
+from system.temporary_directory import TemporaryDirectory
+from validation_workflow.api_test_cases import TestCases
+from validation_workflow.download_utils import DownloadUtils
+from validation_workflow.validation import Validation
+from validation_workflow.validation_args import ValidationArgs
+
+
+class ValidateRpm(Validation, DownloadUtils):
+
+    def __init__(self, args: ValidationArgs) -> None:
+        super().__init__(args)
+        self.base_url = "https://artifacts.opensearch.org/releases/bundle/"
+        self.tmp_dir = TemporaryDirectory()
+
+    def download_artifacts(self) -> bool:
+        architectures = ["x64", "arm64"]
+        for project in self.args.projects:
+            for architecture in architectures:
+                url = f"{self.base_url}{project}/{self.args.version}/{project}-{self.args.version}-linux-{architecture}.rpm"
+                if ValidateRpm.is_url_valid(url) and ValidateRpm.download(url, self.tmp_dir):
+                    logging.info(f"Valid URL - {url} and Download Successful !")
+                else:
+                    logging.info(f"Invalid URL - {url}")
+                    raise Exception(f"Invalid url - {url}")
+        return True
+
+    def installation(self) -> bool:
+        try:
+            os.chdir(self.tmp_dir.path)
+            os.system('sudo rpm --import https://artifacts.opensearch.org/publickeys/opensearch.pgp')
+            for project in self.args.projects:
+                os.system(f'sudo yum remove {project} -y')
+                path = os.path.join(self.tmp_dir.path, project + "-" + self.args.version + "-linux-" + self.args.arch + ".rpm")
+                os.system(f'sudo rpm -ivh {path}')
+        except:
+            raise Exception('Failed to install Opensearch')
+        return True
+
+    def start_cluster(self) -> bool:
+        try:
+            for project in self.args.projects:
+                os.system(f'sudo systemctl enable {project}')
+                os.system(f'sudo systemctl start {project}')
+                time.sleep(20)
+                os.system(f'sudo systemctl status {project}')
+        except:
+            raise Exception('Failed to Start Cluster')
+        return True
```

Comment:
Will this hard fail and exit or just throw an exception and continue the flow?

Reply:
This exits from the flow and raises exception Sayali.